targets/arm/mikroe/stm32/src/tim/implementation_1/hal_ll_tim.c

```diff
@@ -1132,37 +1120,94 @@ static void _hal_ll_tim_alternate_functions_set_state( hal_ll_tim_hw_specifics_m
 
 static void _hal_ll_tim_hw_init( hal_ll_tim_hw_specifics_map_t *map ) {
 
-    uint32_t  ck_psc;
+    uint32_t tim_clk = 0;
+    uint32_t prescaler = 0;
+    uint32_t period = 0;
+    uint32_t ticks = 0;
 
+    /**
+     * Obtain pointer to TIM register block (TIMx)
+     */
     hal_ll_tim_base_handle_t *hal_ll_hw_reg = hal_ll_tim_get_base_struct( map->base );
 
-    // Counter used as upcounter.
+    /**
+     * Configure timer as up-counter (edge-aligned mode)
+     * Clear DIR bit → counter counts from 0 up to ARR
+     */
     clear_reg_bit( &hal_ll_hw_reg->cr1, HAL_LL_TIM_CR1_DIR_BIT );
 
-    // Get clock speed.
-    ck_psc = _hal_ll_tim_get_clock_speed( map->base ) / map->freq_hz;
-
-    // This should not be the case. ck_psc should not be greater than 65535, which is max value.
-    if ( ck_psc > UINT16_MAX ) {
-        // In case of duty period beeing greater than 65535 set ck_psc to max value.
-        ck_psc = UINT16_MAX - 1; // Because of ( ck_psc / UINT16_MAX + 1 ) ), has to be minus 1.
-    } else if ( ck_psc < 1 ) {
-        ck_psc = 1;
+    /**
+     * Get actual timer input clock (already includes APB ×2 rule
```

Comment:
NECTO can't read the `×` symbol, so this line should be changed to
```suggestion
     * Get actual timer input clock (already includes APB x2 rule
```